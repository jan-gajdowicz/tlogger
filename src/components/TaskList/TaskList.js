import React, { Component } from 'react';
import TaskItem from '../TaskItem/TaskItem';
import TaskCreator from '../TaskCreator/TaskCreator';
import './TaskList.css';

class TaskList extends Component {
    constructor(props) {
        super(props);
        this.state = { 
            items: [
                {name: 'task one', time: 300000},
                {name: 'task two', time: 150000},
            ]
        };
    }   
    render() {
        return (
        <div className="task-list">
            <h1>Task List</h1>
            <TaskCreator />
            {this.state.items.map((item, index) => (
                <TaskItem item={item} key={index}/>
            ))}
        </div>
        );
    }
}

export default TaskList;