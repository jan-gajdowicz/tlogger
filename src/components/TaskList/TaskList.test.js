import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import React from 'react'
import TaskList from './TaskList'

const mockStore = configureMockStore({})
const store = mockStore({
  tasks: {
    tasks: [
      {
        taskCreationTime: '2019-10-03T07:05:02.935Z',
        taskID: 'task_1570086302935',
        taskName: 'shopping',
        taskTimeFormatted: '00:00:01',
        taskTimeInMs: 1000
      }
    ]
  }
})

const wrapper = mount(
  <Provider store={store}>
    <TaskList />
  </Provider>
).debug()

it('tests against snapshot', () => {
  expect(wrapper).toMatchSnapshot()
})

// snapshot serializers