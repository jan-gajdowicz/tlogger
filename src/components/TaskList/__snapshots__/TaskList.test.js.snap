// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests against snapshot 1`] = `ReactWrapper {}`;
